packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -0,0 +1,426 @@
+    setShowPreviewer(isPreviewerVisible);
+  }, [setShowPreviewer, isPreviewerVisible]);
+
+  // Previewer Methods
+
+  const progressCallback = useCallback(
+    (index: number) => {
+      return (progress: { loaded: number; total: number }) => {
+        const percentage = Math.floor((progress.loaded / progress.total) * 100);
+        const status = fileStatusesRef.current[index];
+        fileStatusesRef.current[index] =
+          percentage !== 100
+            ? { ...status, percentage, fileState: 'loading' }
+            : { ...status, percentage, fileState: 'success' };
+        const newFileStatuses = [...fileStatusesRef.current];
+        setFileStatuses(newFileStatuses);
+      };
+    },
+    [setFileStatuses]
+  );
+
+  const errorCallback = useCallback(
+    (index: number) => {
+      return (err: string) => {
+        const status = fileStatusesRef.current[index];
+        fileStatusesRef.current[index] = {
+          ...status,
+          fileState: 'error',
+          fileErrors: translate(err.toString()),
+        };
+
+        const newFileStatuses = [...fileStatusesRef.current];
+        setFileStatuses(newFileStatuses);
+        setLoading(false);
+        if (typeof onError === 'function') onError(err);
+      };
+    },
+    [onError, setFileStatuses]
+  );
+
+  const completeCallback = useCallback(() => {
+    return (event: { key: string }) => {
+      if (typeof onSuccess === 'function') onSuccess(event);
+    };
+  }, [onSuccess]);
+
+  const onDelete = () => {
+    //todo delete
+  };
+
+  const onPause = useCallback(
+    (index: number): (() => void) => {
+      return function () {
+        const status = fileStatuses[index];
+        if (isUploadTask(status.uploadTask)) {
+          status.uploadTask.pause();
+        }
+        const newFileStatuses = [...fileStatuses];
+
+        newFileStatuses[index] = { ...status, fileState: 'paused' };
+        setFileStatuses(newFileStatuses);
+      };
+    },
+    [fileStatuses, setFileStatuses]
+  );
+
+  const onResume = useCallback(
+    (index: number): (() => void) => {
+      return function () {
+        const status = fileStatuses[index];
+
+        if (isUploadTask(status.uploadTask)) {
+          status.uploadTask.resume();
+        }
+        const newFileStatuses = [...fileStatuses];
+
+        newFileStatuses[index] = { ...status, fileState: 'resume' };
+        setFileStatuses(newFileStatuses);
+      };
+    },
+    [fileStatuses, setFileStatuses]
+  );
+
+  const onFileClick = useCallback(() => {
+    // start upload
+    setLoading(true);
+    const uploadTasksTemp: UploadTask[] = [];
+    fileStatuses.forEach((status, i) => {
+      if (status?.fileState === 'success') return;
+
+      const uploadTask = uploadFile({
+        file: status.file,
+        fileName: status.name,
+        level,
+        resumable,
+        progressCallback: progressCallback(i),
+        errorCallback: errorCallback(i),
+        completeCallback: completeCallback(),
```

Comment:
Can we just pass `onSuccess` directly here?

Reply:
We'd still need to check if it's a function or not. I could inline that in here, but that doesn't seem much better then having a function that check it.

Reply:
Maybe I'm misreading but doesn't `completeCallback` just to resolve to `() => onSuccess()` which could just be written in the short form `onSuccess`?

Reply:
👍 [38de413](https://github.com/aws-amplify/amplify-ui/pull/3028/commits/38de4137f394f3db251a8df55a4059c821850d22)